### DISCUSSION TITLE
How can this be done using MVUX

### DISCUSSION THREAD

**Discussion by mcNets** [unoplatform/uno#18140] [Q&A] [answered]
I'm working on an app that shows a board with information about a bunch of machines. I add a "Card" for each machine in the system with info about its current status. `NOTE: this list can be filtered.`

That's my current ViewModel:

```csharp
internal partial record MachineBoardViewModel(MachinesService MachinesService)
{
    public IState<MachineCriteria> Criteria => State.Value(this, () => new MachineCriteria());

    public IListFeed<MachinesListItem> Machines => Criteria.SelectAsync(MachinesService.GetMachinesListAsync).AsListFeed();
}
```

Then the user can start a task that connects to each machine sequentially via gRPC, and if the machine has a visible card in the screen, it should update the info without removing and inserting a new card. This task runs until the user stops it.

Is that possible using MVUX?

**Comment by ajpinedam** [comment]
Hey @mcNets, to validate I understood your inquiry:

The Machines `MachineListItem` list will be fetched based on the Criteria `MachineCriteria`. This Machine List is displayed using a `Card` Control (maybe inside an ItemControl or ListView) for each Machine. I have it clear until here, I believe.

Your question is that you want to find a way, using MVUX, so that the new Machines data that will arrive from the gRPC Streaming Tasks is updated in place instead of removing and inserting the new `MachineListItem` into the Collection (the `ListFeed` Machines object), is this correct?

cc:  Adding  @dr1rrb to the conversation, as he could add valuable info here.

**Reply by mcNets** [reply]
hi @ajpinedam, thanks for helping.

Yes you're right, every few seconds each 'MachineListItem' will be updated with new info so I can't use a collection to fill the GridView or ListView.

I need somehow to assign an Item to each control and find a way to update it when new info is coming.

**Reply by francoistanguay** [reply, marked as answer]
First, if you want to update items in a list, they need to be stored in an IListState. You can create a ListState from a Feed.
Once your have your ListState, you can Update your items based on the changes you're fetching.

https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Mvux/ListStates.html#fromfeed

https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Mvux/ListStates.html#update

**Reply by mcNets** [reply]
Sorry @francoistanguay, I haven't said thankyou before.
